### PR TITLE
Purchase flow

### PR DESCRIPTION
**Pull request by SofiaCantero24**
## What does this do?
Purchase flow UI

https://github.com/user-attachments/assets/993ff09d-67a6-4afc-b818-61047e341cc7

## Notion Ticket
* [Purchase Flow](https://www.notion.so/rootstrap/Shopping-Cart-Purchase-flow-Main-Flow-14259347eba68089be17f588304d9a6a?pvs=4)

## Figma view
* [Purchase Flow screen](https://www.figma.com/file/uUWXfDpuvDtnOHdSRjv0t1/Black-Market---iOS?type=design&node-id=942-2295&mode=design&t=2OqGoK3yhEqluy8w-4)

### CODE REVIEW

File: src/components/payment/buy-form.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import { HeaderLogo } from '../header-logo';
+
+export const _adressSchema = z.object({
+  city: z.string({
+    required_error: 'City required',
+  }),
+  country: z.string({
+    required_error: 'Country required',
+  }),
+  adress1: z.string({
```
Comment (fernandatoledo) [inline src/components/payment/buy-form.tsx]: typo, should be address, check the rest of the code as it's a misspelled throughout the code

File: src/app/(app)/_layout.tsx [hunk 2]
```diff
@@ -102,7 +109,8 @@ export default function TabLayout() {
               renderIcon(icon, color, focused),
             tabBarTestID: testID,
             headerShown,
-            tabBarButton: name === 'product-list' ? () => null : undefined,
+            tabBarButton:
+              name === ('product-list' || 'payment') ? () => null : undefined,
```
Comment (fernandatoledo) [inline src/app/(app)/_layout.tsx]: this line of code is doing the exact same thing that the previous one, did you mean


name === 'product-list' || name === 'payment' ? () => null : undefined, ?

File: src/app/(app)/payment.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { BuyForm } from '@/components/payment/buy-form';
+import { SafeAreaView } from '@/ui';
+
+export default function Payment() {
+  // const formMethods = useForm();
```
Comment (fernandatoledo) [inline src/app/(app)/payment.tsx]: Remove commented code

File: src/components/date-picker.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  label: string;
+  value: number;
+};
+
+const monthOptions = Array.from({ length: 12 }, (_, index) => ({
+  label: (index + 1).toString(),
+  value: index + 1,
+}));
+
+const yearOptions = [
```
Comment (fernandatoledo) [inline src/components/date-picker.tsx]: The yearOptions array has hardcoded values, this means you'd have to update this values every year, what if you generate it automatically instead?


```
const currentYear = new Date().getFullYear();
const yearOptions = Array.from({ length: 10 }, (_, i) => ({
  label: (currentYear + i).toString(),
  value: currentYear + i,
}));
```

File: src/components/payment/buy-form.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+              accessibilityLabel="Label 2"
+              className="mb-4"
+            />
+          </View>
+          <Button
+            label="Buy"
+            className="h-12"
+            textClassName="font-bold text-base"
+          />
+          <Button variant="outline" label="Cancel" className="mb-8" />
```
Comment (fernandatoledo) [inline src/components/payment/buy-form.tsx]: shouldn't the form receive the callbacks to pass down to this buttons?

**Review by juanchoperezj — APPROVED** [review]

**Review by fernandatoledo — APPROVED** [review]

**Pull request merged**